src/gui/transferlistwidget.cpp

```diff
@@ -607,6 +607,57 @@ void TransferListWidget::askNewCategoryForSelection()
     } while(invalid);
 }
 
+void TransferListWidget::askAddTagsForSelection()
+{
+    const QStringList tags = askTagsForSelection("Add Tags");
+    foreach (const QString &tag, tags)
+        addSelectionTag(tag);
+}
+
+void TransferListWidget::askRemoveTagsForSelection()
+{
+    const QStringList tags = askTagsForSelection("Remove Tags");
+    foreach (const QString &tag, tags)
+        removeSelectionTag(tag);
+}
+
+void TransferListWidget::confirmRemoveAllTagsForSelection()
+{
+    QMessageBox::StandardButton response;
+    response = QMessageBox::question(
+        this, tr("Remove All Tags"), tr("Remove all tags from selected torrents?"),
+        QMessageBox::Yes | QMessageBox::No);
+    if (response == QMessageBox::Yes)
+        clearSelectionTags();
+}
+
+QStringList TransferListWidget::askTagsForSelection(const QString &dialog_title)
+{
+    QStringList tags;
+    bool invalid;
+    do {
+        bool ok;
+        invalid = false;
+        const QString tags_input = AutoExpandableDialog::getText(
+            this, tr(dialog_title.toStdString().c_str()), tr("Comma-separated tags:"),
+            QLineEdit::Normal, "", &ok).trimmed();
+        if (!ok || tags_input.isEmpty())
+            return QStringList();
+        tags = tags_input.split(',', QString::SkipEmptyParts);
+        for (QString &tag : tags) {
+            tag = tag.trimmed();
+            if (!BitTorrent::Session::isValidTag(tag)) {
+                QString error("Tag name: '");
+                error += tag + "' is invalid.";
+                QMessageBox::warning(this, tr("Invalid tag"), tr(error.toStdString().c_str()));
```

Comment:
`tr(error.toStdString().c_str())` not correct, use
`tr("abc %1 %2").arg("first").arg("second")`